### ISSUE TITLE
HTTP Client 1.0 spec?

### ISSUE THREAD

**Issue by adamretter** [expath/expath-cg#103]
So er... I can't seem to find the spec for the HTTP 1.0 client in this repo?

Anyone know where it is?

**Comment by fgeorges** [comment]
I guess it falls under the "no one took responsibility of it when migrating to W3C" thing.  As it is already used as is, but still in "candidate" status, I suggest to:

- publish a "draft report" on the CG as is
- announce it, say it is very unlikely we accept any change (esp. with the work on 2.0)
- move it to "final" ASAP

**Comment by adamretter** [comment]
@fgeorges why not make things easier on yourself. By just moving it to the CG as "final". It has not changed in a very long time. My PRs for changes were never accepted and merged into the shared implementation, so I guess there is no need to update the spec with those additions and fixes?!?

**Comment by fgeorges** [comment]
PR for changes to the spec?

**Comment by adamretter** [comment]
No. For the "shared implementation". It would have then needed changes to the spec to make things available:

1. chunked support
2. http-version
3. gzip/deflate support
4. etc (I forget exactly what else I added)

**Comment by adamretter** [comment]
@fgeorges Do you plan to publish this as a 1.0 on the W3C CG page?

I would very much like to get it added to https://www.specref.org/

**Comment by fgeorges** [comment]
Well, last time we looked at the PR on the implementation, the code did not compile, because of some missing POM :-)

Yes, I am digging around for the spec.  I don't think it is worth reviving the original formatting toolset, so I am actually considering copy-and-pasting the content to the same authoring tool you use.  It is ReSpec, isn't it?  Do you mind giving me a couple of pointers?  Did you do anything special with the HTTP Client 2.0 module?  So we use the same "config" if any.

Never heard of Specref before but, yeah, why not.

**Comment by fgeorges** [comment]
A first version is available at http://expath.github.io/expath-cg/specs/http-client/.

@adamretter @ChristianGruen Can you have a look?  Especially for layout and conventions.  I followed conventions from your draft of HTTP Client 2.0, the F&O recommendation, and adapted a few things to follow the W3C CG use of ReSpec.

As this is going to be the first publication of an EXPath report using ReSpec, I suggest we give a bit of attention to the ReSpec config, conventions, and formatting, so we can then follow the same in the future.

I am going to add it as a draft report, then once everyone has got a chance to look at it, I'll push it to final.

**Comment by ChristianGruen** [comment]
Hi @fgeorges, that was quick; and the result looks pretty good as far as I can judge!

I see you added some extensions to the Candidate Module from 9 January 2010 (such as the http-version attribute for requests). Do you have a short list of all features that were added in this draft?

**Comment by fgeorges** [comment]
Good catch!  As far as I can tell, this is the only change.  It was sitting somewhere on my "current" draft (on my laptop from back in the days), which had never been published.  Apart from that, there might be a few typos fixed, but not much more.

**Comment by ChristianGruen** [comment]
If that’s the only edit, maybe we could keep the spec identical to the one that is online on expath.org? 

That’s just an idea. Otherwise, we may need to further discuss added features. For example, I am not sure if the version should be added to the response as well? At least that’s what Adam suggested for version 2.0 of the spec.

**Comment by adamretter** [comment]
@fgeorges Initial draft looks good, I have sent a PR for language and linking fixes here - https://github.com/expath/expath-cg/pull/115

**Comment by adamretter** [comment]
@ChristianGruen I would also be in favour of NOT adding the `http-version` feature to 1.0. As it was never on the expath.org version (even though we implemented it in eXist-db).

**Comment by fgeorges** [comment]
@adamretter If you did, and I did, that's a good proportion of the implementations :-)  I suggest I leave it and mark it as a "MAY".  Thanks for the PR, will review it ASAP (probably not before Monday).
